libuvccamera/src/main/java/com/serenegiant/usb/USBMonitor.java

```diff
@@ -251,7 +251,7 @@ public synchronized void unregister() throws IllegalStateException {
     }
 
     public synchronized boolean isRegistered() {
-        return !mDestroyed && (mPermissionIntent != null);
+        return !mDestroyed;
```

Comment:
This function now returns true even if it's not registered, therefore it doesn't work as intended. I checked the usage and found the only caller to be `requestPermission`. So instead of this change, you can replace the check for `isRegistered()` with `!mDestroyed`, it would behave the same.

But `mPermissionIntent` is used within `requestPermission`, so if we won't check it by calling `isRegistered()`, we need to explicitly check for the `mPermissionIntent` not being null before calling `mUsbManager.requestPermission(device, mPermissionIntent);`

I checked the whole function and noticed minor issue - even if the `device` is null it will call `processCancel(device);`.

Here is the whole suggested update:

1. replaced `isRegistered()` with `!mDestroyed`
2. `if (device != null)` move to be the first check and do nothing if it's null
3. check `mPermissionIntent != null` before calling `mUsbManager.requestPermission`

```
    public void requestPermission(final UsbDevice device) {
        if (DEBUG) Log.v(TAG, "requestPermission:device=" + device.getDeviceName());
        synchronized (USBMonitor.class) {
            if (device != null) {
                if (!mDestroyed) {
                    if (mUsbManager.hasPermission(device)) {
                        // call onConnect if app already has permission
                        processOpenDevice(device);
                    } else if (mPermissionIntent != null) {
                        try {
                            // if no usb permission, request permission
                            mUsbManager.requestPermission(device, mPermissionIntent);
                        } catch (final Exception e) {
                            // With Android5.1.x of GALAXY, this action may throw exception:android.permission.sec.MDM_APP_MGMT
                            Log.w(TAG, e);
                            processCancel(device);
                        }
                    } else {
                        processCancel(device);
                    }
                } else {
                    processCancel(device);
                }
            }
        }
    }
```